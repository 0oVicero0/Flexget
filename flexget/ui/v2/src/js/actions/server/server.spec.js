import fetchMock from 'fetch-mock';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import {
  reloadServer,
  shutdownServer,
  dismissReload,
  showShutdownPrompt,
  dismissShutdownPrompt,
  dismissShutdown,
} from 'actions/server';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore();

describe('actions/server', () => {
  afterEach(() => {
    fetchMock.restore();
    store.clearActions();
  });

  describe('reloadServer', () => {
    it('dispatches the correct action on successful server reload', () => {
      fetchMock
        .post('/api/server/manage', 200);

      return store.dispatch(reloadServer())
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('dispatches the correct action on unsuccessful server reload', () => {
      fetchMock
        .post('/api/server/manage', { body: { message: 'error' }, status: 500 });

      return store.dispatch(reloadServer())
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });
  });

  describe('shutdownServer', () => {
    it('dispatches the correct action on successful server shutdown', () => {
      fetchMock
        .post('/api/server/manage', 200);

      return store.dispatch(shutdownServer())
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('dispatches the correct action on unsuccessful server shutdown', () => {
      fetchMock
        .post('/api/server/manage', { body: { message: 'error' }, status: 500 });

      return store.dispatch(shutdownServer())
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });
  });
});
