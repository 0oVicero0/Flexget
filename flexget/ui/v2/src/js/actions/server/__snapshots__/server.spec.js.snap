// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/server dismissReload dispatches the correct action 1`] = `
Object {
  "error": false,
  "meta": Object {
    "ignore": false,
    "message": undefined,
    "namespace": "SERVER",
  },
  "payload": undefined,
  "type": "SERVER_RELOAD_DISMISS",
}
`;

exports[`actions/server dismissShutdown dispatches the correct action 1`] = `
Object {
  "error": false,
  "meta": Object {
    "ignore": false,
    "message": undefined,
    "namespace": "SERVER",
  },
  "payload": undefined,
  "type": "SERVER_SHUTDOWN_DISMISS",
}
`;

exports[`actions/server dismissShutdownPrompt dispatches the correct action 1`] = `
Object {
  "error": false,
  "meta": Object {
    "ignore": false,
    "message": undefined,
    "namespace": "SERVER",
  },
  "payload": undefined,
  "type": "SERVER_SHUTDOWN_PROMOPT_DISMISS",
}
`;

exports[`actions/server reloadServer dispatches the correct action on successful server reload 1`] = `
Array [
  Object {
    "meta": Object {
      "loading": true,
      "namespace": "SERVER",
    },
    "type": "SERVER_RELOAD",
  },
  Object {
    "error": true,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "SERVER",
    },
    "payload": [FetchError: invalid json response body at /api/server/manage reason: Unexpected end of JSON input],
    "type": "SERVER_RELOAD",
  },
]
`;

exports[`actions/server reloadServer dispatches the correct action on unsuccessful server reload 1`] = `
Array [
  Object {
    "meta": Object {
      "loading": true,
      "namespace": "SERVER",
    },
    "type": "SERVER_RELOAD",
  },
  Object {
    "error": true,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "SERVER",
    },
    "payload": [Error: error],
    "type": "SERVER_RELOAD",
  },
]
`;

exports[`actions/server showShutdownPrompt dispatches the correct action 1`] = `
Object {
  "error": false,
  "meta": Object {
    "ignore": false,
    "message": undefined,
    "namespace": "SERVER",
  },
  "payload": undefined,
  "type": "SERVER_SHUTDOWN_PROMPT",
}
`;

exports[`actions/server shutdownServer dispatches the correct action on successful server shutdown 1`] = `
Array [
  Object {
    "meta": Object {
      "loading": true,
      "namespace": "SERVER",
    },
    "type": "SERVER_SHUTDOWN",
  },
  Object {
    "error": true,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "SERVER",
    },
    "payload": [FetchError: invalid json response body at /api/server/manage reason: Unexpected end of JSON input],
    "type": "SERVER_SHUTDOWN",
  },
]
`;

exports[`actions/server shutdownServer dispatches the correct action on unsuccessful server shutdown 1`] = `
Array [
  Object {
    "meta": Object {
      "loading": true,
      "namespace": "SERVER",
    },
    "type": "SERVER_SHUTDOWN",
  },
  Object {
    "error": true,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "SERVER",
    },
    "payload": [Error: error],
    "type": "SERVER_SHUTDOWN",
  },
]
`;
